pkg/applicationmanager/controllers/applicationrestore.go

```diff
@@ -406,7 +406,32 @@ func (a *ApplicationRestoreController) handle(ctx context.Context, restore *stor
 	}
 	switch restore.Status.Stage {
 	case storkapi.ApplicationRestoreStageInitial:
-		// Make sure the namespaces exist
+		if restore.Spec.BackupObjectType == resourcecollector.PxBackupObjectType_virtualMachine {
+			// The backuObjectType is VirtualMachine. We need to remove resources such as PVC,
+			// configMap, secret that are not associated to any VirtualMachine (aka orphaned resources).
+			logrus.Infof("It is restore of  VirtualMachine  Backup Object type. Filtering resources not associated to any of the selected VirtualMachines")
+			includeObjects, err := a.filterOrphanedResourcesForVMRestore(restore)
+			if err != nil {
+				message := fmt.Sprintf("error processing VirtualMachine restore : %v", err)
+				log.ApplicationRestoreLog(restore).Errorf(message)
+				a.recorder.Event(restore,
+					v1.EventTypeWarning,
+					string(storkapi.ApplicationRestoreStatusFailed),
+					message)
+				if _, ok := err.(*volume.ErrStorageProviderBusy); ok {
```

Comment:
As as are not accessing any storage here, do we need this error msg `ErrStorageProviderBusy` type? Mainly this are used in the case where storage provider threw an error Eg during backup or restore.